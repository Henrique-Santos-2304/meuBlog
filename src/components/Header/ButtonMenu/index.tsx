import { Button } from "@chakra-ui/react";
import { DragHandleIcon } from "@chakra-ui/icons";

type ButtonProps = { onOpen?: () => void };

const ButtonMenu = ({ onOpen }: ButtonProps) => (
  <Button
    aria-label="Botão para abrir um menu de opções"
    title="Abra o Menu"
    onClick={onOpen}
    padding="2.1rem 1rem"
    mr="10px"
    bg="transparent"
    _hover={{ color: "white.300" }}
    _focus={{ border: "none" }}
    rounded="full"
  >
    <DragHandleIcon
      transition="all .5s linear"
      color="gray.100"
      width="22px"
      height="22px"
      _hover={{ color: "blackAlpha.800" }}
    />
  </Button>
);

export default ButtonMenu;
