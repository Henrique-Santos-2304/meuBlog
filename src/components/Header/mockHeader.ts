export const mockHeader = {
  dataHeader: {
    alt: "Minha logo",
    title: "Esse sou eu ",
    imageLogo: {
      url: "https://res.cloudinary.com/defnibbpl/image/upload/v1635695141/a61237_orkfde_816831fe5f.png",
    },
    preText: "web",
    proText: "dev",
  },
};
