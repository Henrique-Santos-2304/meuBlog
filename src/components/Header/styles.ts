import styled, { css } from "styled-components";
import media from "styled-media-query";

export const ContentBody = styled.div`
  max-width: ${({ theme }) => theme.grid.container};
  height: 8rem;
  background: ${({ theme }) => theme.colors.mainBg};
  position: relative;
  ${media.lessThan("medium")`
    width: 100vw;
  `}
`;
export const Wrapper = styled.header`
  ${({ theme }) => css`
    position: relative;
    margin: 0 auto;
    height: 8rem;
    max-width: ${theme.grid.containerContent};
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0.2rem 3rem;
    background: ${theme.colors.mainBg};
    ${media.lessThan("medium")`
    max-width: 80rem;
    padding: 0.2rem 3rem;

  `}
    ${media.lessThan("small")`
    padding: 0.2rem 2rem;
  `}
  `}
`;
export const ContentMenuFull = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;
    max-width: ${theme.grid.containerContent};
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;

    @media (max-width: 748px) {
      max-width: ${theme.grid.container};
      position: absolute;
      top: 7rem;
      right: 0;
    }
  `};
`;
