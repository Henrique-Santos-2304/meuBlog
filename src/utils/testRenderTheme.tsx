import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
