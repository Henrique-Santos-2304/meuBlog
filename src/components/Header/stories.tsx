import { Story, Meta } from "@storybook/react";
import Header from ".";

export default {
  title: "Header",
  component: Header,
} as Meta;

export const Basic: Story = (args) => <Header {...args} check={false} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <Header {...args} check={false} />;
