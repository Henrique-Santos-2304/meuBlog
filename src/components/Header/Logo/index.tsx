import * as S from "./styles";
import { Box } from "@chakra-ui/react";

const Logo = () => (
  <S.Wrapper>
    <Box padding="4" maxW="3xl">
      <S.PreText aria-label="logo dev">
        Web<span>Dev</span>
      </S.PreText>
    </Box>
  </S.Wrapper>
);

export default Logo;
