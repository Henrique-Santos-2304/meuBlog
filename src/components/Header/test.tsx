import { renderTheme } from "utils/testRenderTheme";
import { screen, fireEvent } from "@testing-library/react";
import Header from ".";

describe("Header", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Header />);
    expect(container).toMatchSnapshot();
  });
  it("should render ", () => {
    renderTheme(<Header />);

    // screen.logTestingPlaygroundURL();
    const menuNav = screen.getByRole("navigation", { hidden: true });
    const button = screen.getByRole("button", { hidden: true });
    expect(button).toBeInTheDocument();
    expect(menuNav.getAttribute("aria-hidden")).toBe("true");
    expect(menuNav).toHaveStyle({ opacity: 0 });

    fireEvent.click(button);
    expect(menuNav.getAttribute("aria-hidden")).toBe("false");
    expect(menuNav).toHaveStyle({ opacity: 1 });
  });
});
