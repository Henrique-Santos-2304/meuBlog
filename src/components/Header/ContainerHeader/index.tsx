import { Container, Flex, useDisclosure } from "@chakra-ui/react";
import ButtonMenu from "../ButtonMenu";
import Menu from "components/Menu";
import Logo from "../Logo";

const ContainerHeader = () => {
  const { isOpen, onOpen, onClose } = useDisclosure();
  return (
    <Container
      position="fixed"
      top="0"
      left="0"
      maxW="full"
      p="0"
      bgGradient="linear(to-l,#a8eb12 ,#00bf72 ,#008793 ,#004d7a 45%, #051937)"
    >
      <Container
        p="0"
        maxW="container.md"
        direction="row"
        centerContent
        height="8rem"
      >
        <Flex
          w="100%"
          h="100%"
          alignItems="center"
          px="2rem"
          justifyContent="space-between"
        >
          <Logo />
          <ButtonMenu onOpen={onOpen} />

          <Menu onClose={onClose} isOpen={isOpen} />
        </Flex>
      </Container>
    </Container>
  );
};
export default ContainerHeader;
