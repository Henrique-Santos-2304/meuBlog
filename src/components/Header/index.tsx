import Logo from "components/Logo";
import { useState } from "react";
import MenuIcon from "components/MenuIcon";
import * as S from "./styles";
import MenuLink from "components/MenuLink";

interface CheckProps {
  check: boolean;
  handleOpenMenu?: () => void;
  isOpen?: boolean;
}
const Header = ({ check = false }: CheckProps) => {
  const [checkMenu, setCheckMenu] = useState(check);
  function handleIsOpenMenu() {
    setCheckMenu(!checkMenu);
  }
  return (
    <>
      <S.ContentBody>
        <S.Wrapper>
          <Logo />
          <S.ContentMenuFull>
            <MenuLink isOpen={checkMenu} />
          </S.ContentMenuFull>
          <MenuIcon check={checkMenu} handleOpenMenu={handleIsOpenMenu} />
        </S.Wrapper>
      </S.ContentBody>
    </>
  );
};
export default Header;
