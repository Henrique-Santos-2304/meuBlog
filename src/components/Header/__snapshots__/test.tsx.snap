// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render  1`] = `
.c2 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.8rem 1.6rem;
  background: #12131C;
  border-radius: 1.2rem;
}

.c3 {
  font-family: 'Tangerine',cursive;
  font-size: 5rem;
  color: #E5E7EB;
}

.c4 {
  font-family: 'Tangerine',cursive;
  font-size: 5rem;
  color: #C11010;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  width: 4.125rem;
  height: 4.125rem;
  border-radius: 50%;
  background: #12131C;
  border: 0.155rem solid transparent;
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
}

.c8:hover {
  width: 4.65rem;
  height: 4.65rem;
}

.c9 {
  display: block;
  width: 1.8rem;
  margin: 0 1rem;
  height: 0.2rem;
  position: relative;
  -webkit-transition: 0.4s ease-in-out;
  transition: 0.4s ease-in-out;
  background: #E5E7EB;
}

.c9::after {
  background: #E5E7EB;
  bottom: -0.6rem;
}

.c9::before {
  background: #E5E7EB;
  top: -0.6rem;
}

.c9::after,
.c9::before {
  content: "";
  display: block;
  width: 100%;
  height: 100%;
  position: absolute;
  -webkit-transition: 0.4s ease-in-out;
  transition: 0.4s ease-in-out;
}

.c0 {
  max-width: 100vw;
  height: 8rem;
  background: #12131C;
  position: relative;
}

.c1 {
  position: relative;
  margin: 0 auto;
  height: 8rem;
  max-width: 86vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.2rem 3rem;
  background: #12131C;
}

.c5 {
  margin: 0 auto;
  max-width: 86vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
}

.c6 {
  width: 86vw;
  padding: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  box-shadow: 0 0 30px #C11010,0 0 10px #E5E7EB;
  -webkit-backdrop-filter: blur(0.1rem);
  backdrop-filter: blur(0.1rem);
  overflow: hidden;
  opacity: 0;
  -webkit-transition: 1.2s ease-in-out;
  transition: 1.2s ease-in-out;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
}

.c7 {
  color: #E5E7EB;
  font-weight: 700;
  padding: 0.8rem;
  font-size: 2.0rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  position: relative;
  -webkit-transition: filter .3s;
  transition: filter .3s;
}

.c7:hover::after {
  content: "";
  position: absolute;
  bottom: -0.1rem;
  right: 0;
  height: 0.3rem;
  background: #C11010;
  -webkit-animation: hoverAnimation .4s forwards;
  animation: hoverAnimation .4s forwards;
}

@media (max-width:450px) {
  .c3 {
    font-size: 3rem;
  }
}

@media (max-width:450px) {
  .c4 {
    font-size: 3rem;
  }
}

@media (max-width:768px) {
  .c0 {
    width: 100vw;
  }
}

@media (max-width:768px) {
  .c1 {
    max-width: 80rem;
    padding: 0.2rem 3rem;
  }
}

@media (max-width:450px) {
  .c1 {
    padding: 0.2rem 2rem;
  }
}

@media (max-width:748px) {
  .c5 {
    max-width: 100vw;
    position: absolute;
    top: 7rem;
    right: 0;
  }
}

@media (max-width:748px) {
  .c6 {
    width: 100vw;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    background: #E5E7EB;
    width: 100vw;
    height: 80vh;
    border-radius: 0;
  }

  .c6 a {
    color: #12131C;
    font-size: 2.5rem;
    padding: 2.4rem 0.8rem;
  }
}

@media (max-width:1020px) {
  .c7 {
    font-size: 1.6rem;
  }
}

<div>
  <div
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
        >
          Web
        </p>
        <p
          class="c4"
        >
          Dev
        </p>
      </div>
      <div
        class="c5"
      >
        <nav
          aria-hidden="true"
          class="c6"
          data-testid="menuLinks"
        >
          <a
            class="c7"
            href="#"
          >
            Inicio
          </a>
          <a
            class="c7"
            href="#project"
          >
            Projetos
          </a>
          <a
            class="c7"
            href="#skills"
          >
            Habilidades
          </a>
          <a
            class="c7"
            href="#about"
          >
            Sobre
          </a>
        </nav>
      </div>
      <div
        class="c8"
        data-testid="divClick"
      >
        <span
          class="c9"
          data-testid="span-test"
        />
      </div>
    </header>
  </div>
</div>
`;
